--- nagvis.ini.php.orig	2012-02-11 10:34:01.382766983 +0100
+++ nagvis.ini.php	2012-02-11 11:22:21.593241959 +0100
@@ -207,15 +207,15 @@
 ; in the share/nagvis/styles directory)
 ;stylesheet=""
 ; target for the icon links
-;urltarget="_self"
+urltarget="_self"
 ; URL template for host object links
-;hosturl="[htmlcgi]/status.cgi?host=[host_name]"
+hosturl="/check_mk/view.py?view_name=host&site=default&host=[host_name]"
 ; URL template for hostgroup object links
-;hostgroupurl="[htmlcgi]/status.cgi?hostgroup=[hostgroup_name]"
+hostgroupurl="/check_mk/view.py?view_name=hostgroup&hostgroup=[hostgroup_name]"
 ; URL template for service object links
-;serviceurl="[htmlcgi]/extinfo.cgi?type=2&host=[host_name]&service=[service_description]"
+serviceurl="/check_mk/view.py?view_name=service&site=default&service=[service_description]&host=[host_name]"
 ; URL template for servicegroup object links
-;servicegroupurl="[htmlcgi]/status.cgi?servicegroup=[servicegroup_name]&style=detail"
+servicegroupurl="/check_mk/view.py?view_name=servicegroup&servicegroup=[servicegroup_name]"
 ; URL template for nested map links
 ;mapurl="[htmlbase]/index.php?mod=Map&act=view&show=[map_name]"

